### PR TITLE
feat(modal): finish modal component

### PR DESCRIPTION
**Pull request by AngeliaQ**
## Checklist

- [x] Fix linting errors
- [x] Tests have been added / updated (or snapshots)

### CODE REVIEW

**Comment by AngeliaQ** [comment]
Task linked: [CU-3tvrd7 Modal component](https://app.clickup.com/t/3tvrd7)

File: components/modal/modal-subtitle.tsx [hunk 1]
```diff
@@ -23,17 +23,18 @@ const ModalSubtitle: React.FC<ModalSubtitleProps> = ({ className, children, ...p
       </p>
       <style jsx>{`
         p {
-          font-size: 0.875rem;
-          line-height: 1.6;
+          font-size: 1rem;
+          line-height: 20px;
```
Comment (yqrashawn) [inline components/modal/modal-subtitle.tsx]: use rem here

File: components/modal/modal-title.tsx [hunk 2]
```diff
@@ -23,18 +23,19 @@ const ModalTitle: React.FC<ModalTitleProps> = ({ className, children, ...props }
       </h2>
       <style jsx>{`
         h2 {
-          font-size: 1.5rem;
-          line-height: 1.6;
-          font-weight: normal;
+          font-size: 1.2857rem;
+          line-height: 25px;
```
Comment (yqrashawn) [inline components/modal/modal-title.tsx]: use rem here

File: components/modal/__tests__/use-imperative-modal.test.tsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    expectModalIsOpened(wrapper)
+
+    wrapper.setProps({ show: false })
+    await updateWrapper(wrapper, 500)
+    expectModalIsClosed(wrapper)
+  })
+})
+
+describe('closeModal', () => {
+  it('should close directly when open is underfined', async () => {
```
Comment (yqrashawn) [inline components/modal/__tests__/use-imperative-modal.test.tsx]: `underfined` typo here

File: components/modal/__tests__/use-imperative-modal.test.tsx [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    await updateWrapper(wrapper, 300)
+    expectModalIsOpened(wrapper)
+
+    wrapper.setProps({ show: false })
+    await updateWrapper(wrapper, 500)
+    expectModalIsClosed(wrapper)
+  })
+})
+
+describe('closeModal', () => {
```
Comment (yqrashawn) [inline components/modal/__tests__/use-imperative-modal.test.tsx]: better put these tests in to a single `describe`

File: components/modal/__tests__/use-imperative-modal.test.tsx [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    wrapper.setProps({ show: true })
+    await updateWrapper(wrapper, 500)
+    wrapper.find('.backdrop').simulate('click', nativeEvent)
+    await updateWrapper(wrapper, 500)
+    expectModalIsClosed(wrapper)
+  })
+})
+
+describe('setVisible', () => {
+  it('when setVisible set underfined, should close directly', async () => {
```
Comment (yqrashawn) [inline components/modal/__tests__/use-imperative-modal.test.tsx]: `underfined` typo

File: components/modal/__tests__/use-imperative-modal.test.tsx [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+        </Modal>
+      )
+    }
+
+    const wrapper = mount(<MockModal />)
+    wrapper.setProps({ show: true })
+    await updateWrapper(wrapper, 500)
+    wrapper.find('.backdrop').simulate('click', nativeEvent)
+    await updateWrapper(wrapper, 500)
+    expectModalIsClosed(wrapper)
```
Comment (yqrashawn) [inline components/modal/__tests__/use-imperative-modal.test.tsx]: duplicate test

File: components/modal/__tests__/use-imperative-modal.test.tsx [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+  })
+})
+
+describe('getVisible', () => {
+  it('should get current visible', async () => {
```
Comment (yqrashawn) [inline components/modal/__tests__/use-imperative-modal.test.tsx]: seems we can simply merge this test into the first one

File: components/modal/modal-action.tsx [hunk 8]
```diff
@@ -1,84 +1,60 @@
...
+  loading: false,
+  shadow: false,
+  auto: false,
+  effect: true,
   disabled: false,
+  className: '',
 }
 
 type NativeAttrs = Omit<React.ButtonHTMLAttributes<any>, keyof Props>
 export type ModalActionProps = Props & typeof defaultProps & NativeAttrs
```
Comment (yqrashawn) [inline components/modal/modal-action.tsx]: export type without type of defaultProps

File: components/modal/modal-icon.tsx [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+  size?: number
+  color?: string
+  onClick?: (event: React.MouseEvent) => void
+}
+
+const defaultProps = {
+  size: 14,
+}
+
+export type ModalIconProps = Props & typeof defaultProps
```
Comment (yqrashawn) [inline components/modal/modal-icon.tsx]: export type without typeof defaultProps

File: components/modal/use-modal-handle.ts [hunk 10]
```diff
@@ -0,0 +1,18 @@
+import { useRef, RefObject } from 'react'
+import { ModalHandles } from './modal-context'
+
+const useModalHandle = () => {
+  // const [visible, setVisible, currentRef] = useCurrentState<boolean>(initialVisible)
```
Comment (yqrashawn) [inline components/modal/use-modal-handle.ts]: remove unused line

**Review by yqrashawn — CHANGES_REQUESTED** [review]

File: components/modal/__tests__/use-imperative-modal.test.tsx (hunk 6 above)
Comment (AngeliaQ) [reply components/modal/__tests__/use-imperative-modal.test.tsx]: it test passing setVisible defaulValue visible=undefined

File: components/modal/modal-subtitle.tsx (hunk 1 above)
Comment (AngeliaQ) [reply components/modal/modal-subtitle.tsx]: done

File: components/modal/modal-title.tsx (hunk 2 above)
Comment (AngeliaQ) [reply components/modal/modal-title.tsx]: done

File: components/modal/__tests__/use-imperative-modal.test.tsx (hunk 3 above)
Comment (AngeliaQ) [reply components/modal/__tests__/use-imperative-modal.test.tsx]: done

File: components/modal/__tests__/use-imperative-modal.test.tsx (hunk 4 above)
Comment (AngeliaQ) [reply components/modal/__tests__/use-imperative-modal.test.tsx]: done

File: components/modal/__tests__/use-imperative-modal.test.tsx (hunk 5 above)
Comment (AngeliaQ) [reply components/modal/__tests__/use-imperative-modal.test.tsx]: done

File: components/modal/__tests__/use-imperative-modal.test.tsx (hunk 7 above)
Comment (AngeliaQ) [reply components/modal/__tests__/use-imperative-modal.test.tsx]: done

File: components/modal/modal-action.tsx (hunk 8 above)
Comment (AngeliaQ) [reply components/modal/modal-action.tsx]: done

File: components/modal/modal-icon.tsx (hunk 9 above)
Comment (AngeliaQ) [reply components/modal/modal-icon.tsx]: done

File: components/modal/use-modal-handle.ts (hunk 10 above)
Comment (AngeliaQ) [reply components/modal/use-modal-handle.ts]: done

**Review by yqrashawn — APPROVED** [review]

**Review by GuichiZhao — APPROVED** [review]

**Pull request merged**